SCons: Move build progress related logic out of main SConstruct

Xrayez opened the pull request:
Moved to methods.py as a `show_progress` method.

Some required changes are made:
- locally imported SCons-specific packages within the method;
- `global` variables converted to `nonlocal` (used in nested functions).

I think this should improve readability/navigation, quite important for the root script IMO.

akien-mga commented:
The AppVeyor error is puzzling: https://ci.appveyor.com/project/akien-mga/godot/builds/31615962

Likely not related to this PR, but the change to SConscript triggered a full rebuild which picked up this warning.

Xrayez commented:
AppVeyor build succeeded this time, but Travis is still pesky due to maintenance as you said (same error on macOS the first time I pushed this PR).

Commit:
Moved to methods.py as a `show_progress` method.

Some required changes are made:
- locally imported SCons-specific packages within the method;
- `global` variables converted to `nonlocal` (used in nested functions).

akien-mga commented:
Thanks!
